tools/duplicate_function_eliminator.py

```diff
@@ -0,0 +1,428 @@
+
```

Comment:
There are some minor changes to the Minifier class in the duplicate_function_eliminator. Maybe I'll remove the Minifier class from duplicate_function_eliminator.py. I'd like to avoid making changes to js_optimizer, so if you're okay with it, I'll have a separate minify_shell function in duplicate_function_eliminator.py.